xkafka/consumer.go

```diff
@@ -77,28 +84,33 @@ func (c *Consumer) Start(ctx context.Context) error {
 
 	c.handler = c.concatMiddlewares(c.handler)
 
-	st := stream.New().WithMaxGoroutines(c.config.concurrency)
+	if c.config.concurrency > 1 {
+		return c.runAsync(ctx)
+	}
+
+	return c.runSequential(ctx)
+}
+
+func (c *Consumer) runSequential(ctx context.Context) error {
 	errChan := make(chan error, 1)
```

Comment:
Can you eliminate the use of `errChan`, I feel that the code will be simpler without it.